### PR TITLE
nrf_802154: Update revision of nrf_802154

### PR DESCRIPTION
**Pull request by ahasztag**
This PR updates revision of the nrf_802154 component to commit 122a8380f9d58cc1fccd2736c917769bdae72c5d of the sdk-nrf-802154

### CODE REVIEW

**Review by czeslawmakarski — APPROVED** [review]

**Review by jciupis — APPROVED** [review]

File: nrf_802154/doc/CHANGELOG.rst [hunk 1]
```diff
@@ -10,6 +10,14 @@ Changelog
 All notable changes to this project are documented in this file.
 See also :ref:`nrf_802154_limitations` for permanent limitations.
 
+Main branch - nRF 802.15.4 Radio Driver
+***************************************
+
+Notable changes
+===============
+
+* Added the  possibility to disable the continuous and modulated carrier functions by setting the NRF_802154_CARRIER_FUNCTIONS_ENABLED define to 0.
```
Comment (FrancescoSer) [inline nrf_802154/doc/CHANGELOG.rst:19]:
```suggestion
* Added the possibility to disable the continuous and modulated carrier functions by setting the NRF_802154_CARRIER_FUNCTIONS_ENABLED define to 0.
```

**Review by FrancescoSer — APPROVED** [review]
Approved by please fix the small issue highlighted in the comment

File: nrf_802154/doc/CHANGELOG.rst (hunk 1 above)
Comment (ahasztag) [reply nrf_802154/doc/CHANGELOG.rst]: Done

**Review by ankuns — CHANGES_REQUESTED** [review]
Bad sha in commit message:
there is no `7f5cb04aa26937d61995d91c0439b70f35e14186` in sdk-nrf-802154,
shouldn't it be `122a8380f9d58cc1fccd2736c917769bdae72c5d` ?
The same in manifest PR https://github.com/nrfconnect/sdk-nrf/pull/10275

**Comment by ahasztag** [comment]
> Bad sha in commit message: there is no `7f5cb04aa26937d61995d91c0439b70f35e14186` in sdk-nrf-802154, shouldn't it be `122a8380f9d58cc1fccd2736c917769bdae72c5d` ? The same in manifest PR [nrfconnect/sdk-nrf#10275](https://github.com/nrfconnect/sdk-nrf/pull/10275)

Fixed

**Review by ankuns — APPROVED** [review]

**Pull request merged**